src/client/common/installer/productInstaller.ts

```
@@ -346,6 +352,46 @@ export class RefactoringLibraryInstaller extends BaseInstaller {
 }
 
 export class DataScienceInstaller extends BaseInstaller {
+    // Override base installer to support a more DS-friendly streamlined installation.
+    public async install(
+        product: Product,
+        interpreterUri?: InterpreterUri,
+        cancel?: CancellationToken
+    ): Promise<InstallerResponse> {
+        // Get a list of known installation channels, pip, conda, etc.
+        const channels: IModuleInstaller[] = await this.serviceContainer
+            .get<IInstallationChannelManager>(IInstallationChannelManager)
+            .getInstallationChannels();
+
+        // Assume the Uri is actually a full PythonInterpreter...
+        const interpreter = interpreterUri as PythonInterpreter;
+
+        // Pick an installerModule based on whether the interpreter is conda or not.
+        let installerModule;
+        if (!interpreter || !interpreter.hasOwnProperty('type')) {
+            return InstallerResponse.Ignore; // All data science packages require an interpreter be passed in.
```

Thanks @IanMatthewHuff 
This should never happen, hence we should throw an ugly error here.
Basically the program is misbehaving and we need to be aware of this.

Instead of using `const interpreter = interpreterUri as PythonInterpreter;`
You can use the following:
```typescript
if (isResource(interpreterUri)){
	// This doesn't really need to be a very userfriendly error.
	// This shouldn't even occur in production.
	throw new Error('All data science packages require an interpreter be passed in');
}

// At this point `interpreterUri` is of type PythonInterpreter
// There is not need to cast it
```

---

I agree with you guys.